Proposal: Allow specification of LABEL Name/Value pairs in image json content

rhatdan opened the pull request:
```
Save "UserData" field in Dockerfile into image content.

This will allow a user to save user data into an image, which
can later be retrieved using:

docker inspect IMAGEID

I have copied this from the "Comment" handling in docker images.

We want to be able to add Name/Value data to an image to describe the image,
and then be able to use other tools to look at this data, to be able to do
security checks based on this data.

We are thinking about adding version names,
Perhaps listing the content of the dockerfile.
Descriptions of where the code came from etc.

This USERDATA field should also be allowed to be specified in the
docker import --change USERDATA:Name=Value
docker commit --change USERDATA:Name=Value

Docker-DCO-1.1-Signed-off-by: Dan Walsh <dwalsh@redhat.com> (github: rhatdan)
```

ibuildthecloud commented:
@rhatdan Just a heads up that I've create a PR https://github.com/docker/docker/pull/9882 that is essentially a follow on to this PR.  In short, I like this approach with the slight change the `USERDATA` becomes `LABEL`.  The PR I've created is just an extension of this work but adding labels to containers to in a fashion very similar to `ENV` and `-e`.

In the ideal world this PR would be accepted and then my PR would get merged after that.

ibuildthecloud commented on docs/sources/reference/api/docker_remote_api.md:
```diff
@@ -175,6 +175,10 @@ Added a `pause` parameter (default `true`) to pause the container during commit
 `POST /build`
 
 **New!**
+Build now has support for `LABEL` command which can be used to add user data
```
Can you move this to the v1.17 section

ibuildthecloud commented on docs/sources/reference/api/docker_remote_api_v1.16.md:
```diff
@@ -1329,6 +1330,11 @@ Create a new image from a container's changes
+                     "Vendor=Acme",
+                     "License=GPL",
+                     "Version=1.0"
+             ],
```
Can you move this to the docker_remote_api_v1.17.md

ibuildthecloud commented on runconfig/config.go:
```diff
@@ -33,6 +33,8 @@ type Config struct {
 	MacAddress      string
 	OnBuild         []string
+	SecurityOpt     []string
+	Label           []string
```
I feel Labels would be more consistent with other fields such as ExposedPorts and Volumes.  I know env is not plural, but I think thats just because "envs" is weird.

ibuildthecloud commented on docs/sources/reference/api/docker_remote_api_v1.17.md:
```diff
@@ -129,6 +129,11 @@ Create a container
              ],
              "Entrypoint": "",
              "Image": "base",
+	     "Labels": [
```
Typo, should be { ... } not [ ... ]

ibuildthecloud commented on runconfig/merge.go:
```diff
@@ -84,6 +84,32 @@ func Merge(userConf, imageConf *Config) error {
 		}
 	}
 
+	if len(userConf.PortSpecs) > 0 {
```
Why is this code for PortSpecs being added?

ibuildthecloud commented on docs/sources/reference/api/docker_remote_api.md:
```diff
@@ -49,6 +49,12 @@ You can still call an old version of the API using
 `POST /containers/(id)/attach` and `POST /exec/(id)/start`
 
 **New!**
+Build now has support for `LABEL` command which can be used to add user data
```
I think this block needs to move above `POST /containers/(id)/attach` and `POST /exec/(id)/start` and then the line `POST /build` should be added above it.  Looks like the format is API method/path followed by a paragraph describing the change to the API.

rhatdan replied:
Moved it.

rhatdan committed:
Allow specification of Label Name/Value pairs in image json content

Save "LABEL" field in Dockerfile into image content.

This will allow a user to save user data into an image, which
can later be retrieved using:

docker inspect IMAGEID

I have copied this from the "Comment" handling in docker images.

We want to be able to add Name/Value data to an image to describe the image,
and then be able to use other tools to look at this data, to be able to do
security checks based on this data.

We are thinking about adding version names,
Perhaps listing the content of the dockerfile.
Descriptions of where the code came from etc.

This LABEL field should also be allowed to be specified in the
docker import --change LABEL:Name=Value
docker commit --change LABEL:Name=Value

Docker-DCO-1.1-Signed-off-by: Dan Walsh <dwalsh@redhat.com> (github: rhatdan)

icecrime commented:
@rhatdan I'm trying to push #9882 forward: are you ok with me closing that one?

rhatdan commented:
@icecrime

rhatdan commented:
@icecrime yes.
